src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -682,20 +683,20 @@ trait TypeDiagnostics {
             else if (sym.isMethod) s"method ${sym.name.decoded}"
             else if (sym.isModule) s"object ${sym.name.decoded}"
             else "term"
-            )
+          )
           typer.context.warning(pos, s"$why $what in ${sym.owner} $cond")
         }
+        def typeWarning(defn: SymTree): Unit = {
+          val why = if (defn.symbol.isPrivate) "private" else "local"
+          typer.context.warning(defn.pos, s"$why ${defn.symbol.fullLocationString} is never used")
+        }
+
+        for (defn <- unusedPrivates.unusedTerms if shouldWarnOn(defn.symbol)) { termWarning(defn) }
+        for (defn <- unusedPrivates.unusedTypes if shouldWarnOn(defn.symbol)) { typeWarning(defn) }
```

Comment:
Now it's two one-liners; hopefully that's pleasing enough to the eye.

Reply:
Let me go get my computer glasses.